api/src/test/java/org/openmrs/util/OpenmrsUtilTest.java

```diff
@@ -755,4 +761,172 @@ public void copyFile_shouldCopyInputstreamToOutputstreamAndCloseTheOutputstream(
 		verify(output, times(1)).close();
 	}
 
+	/**
+	 * @see OpenmrsUtil#isTestMode()
+	 */
+	@Test
+	public void isTestMode_shouldPassIfFunctionalTestModeIsEnabled() {
+		
+		System.setProperty("FUNCTIONAL_TEST_MODE", Boolean.toString(true));
+		boolean isTestMode = OpenmrsUtil.isTestMode();
+
+		assertTrue(isTestMode);
+	}
+
+	/**
+	 * @see OpenmrsUtil#isTestMode()
+	 */
+	@Test
+	public void isTestMode_shouldPassIfFunctionalTestModeIsDisabled() {
+
+		System.setProperty("FUNCTIONAL_TEST_MODE", Boolean.toString(false));
+		boolean isTestMode = OpenmrsUtil.isTestMode();
+
+		assertFalse(isTestMode);
+	}
+
+	/**
+	 * @see OpenmrsUtil#compareLists(Collection, Collection)
+	 */
+	@Test
+	public void compareLists_shouldPassIfMapOfDifferencesIsCorrect() {
+		
+		LinkedList<Integer> origList = new LinkedList<>(Arrays.asList(1,2,3));
+		LinkedList<Integer> newList = new LinkedList<>(Arrays.asList(2,3,4));
```

Comment:
can you please create a list with all elements you expect to be returned by the function? I think this would make the test easier to read. You might even be able to inline the result in the assertion:

and for asserting the result is correct, http://hamcrest.org/JavaHamcrest/javadoc/1.3/ might help (assertions like assertThat(result.size(), is() combined with assertThat(result, hasItems("2","3", "4")))